assets/js/components/settings/SettingsApp.js

```diff
@@ -44,62 +44,57 @@ import { Cell, Grid, Row } from '../../material-components';
 
 const { useSelect, useDispatch } = Data;
 
-export default function SettingsApp() {
-	const [ hash, setHash ] = useHash();
-
-	const parseHash = ( hashToParse ) => hashToParse.replace( '#', '' ).split( /\// );
+const hashFrom = ( tabID, slug, state ) => {
+	const fragments = [ tabID ];
 
-	let [ activeTabID, moduleSlug, moduleState ] = parseHash( hash );
-	if ( ! activeTabID ) {
-		activeTabID = 'settings';
+	if ( tabID === 'settings' && slug && state !== 'closed' ) {
+		fragments.push( slug );
+		if ( state ) {
+			fragments.push( state );
+		}
 	}
 
-	const isModuleActive = useSelect( ( select ) => moduleSlug ? select( CORE_MODULES ).isModuleActive( moduleSlug ) : null );
+	return fragments.join( '/' );
+};
 
-	moduleSlug = isModuleActive && moduleSlug;
-	moduleState = useSelect( ( select ) => moduleSlug ? select( CORE_MODULES ).getModuleSettingsPanelState( moduleSlug ) : null );
+const parseHash = ( hashToParse ) => hashToParse.replace( '#', '' ).split( /\// );
 
+export default function SettingsApp() {
+	const [ hash, setHash ] = useHash();
+	const isFirstMount = useFirstMountState();
 	const { setModuleSettingsPanelState } = useDispatch( CORE_MODULES );
-
-	if ( moduleSlug && ! moduleState ) {
-		moduleState = 'view';
-	}
-
+	const [ initialActiveTabID, initialModuleSlug, initialModuleState ] = parseHash( hash );
+	const [ activeTabID, setActiveTabID ] = useState( initialActiveTabID || 'settings' );
+	const [ moduleSlug, setModuleSlug ] = useState( initialModuleSlug );
 	const activeTab = SettingsApp.tabToIndex[ activeTabID ];
 
-	const hashFrom = ( tabID, slug = moduleSlug, state = moduleState ) => {
-		const fragments = [ tabID ];
-
-		if ( tabID === 'settings' && slug && state !== 'closed' ) {
-			fragments.push( slug );
-			if ( state ) {
-				fragments.push( state );
-			}
+	if ( isFirstMount ) {
+		if ( initialModuleSlug ) {
+			setModuleSettingsPanelState( initialModuleSlug, initialModuleState );
```

Comment:
This is the important change here. `useEffect` runs too late and `moduleState` would otherwise be `closed` on the initial render, even when properly parsed as `view` or `edit` from the hash before this. This will essentially run synchronously but only on the very first render. Since `SettingsApp` is never unmounted, this only happens once. It's all somewhat temporary though as most of this will change again once we add the router.